test/end-to-end/oauth2provider.test.js

```diff
@@ -167,6 +168,67 @@ describe("SuperTokens OAuth2Provider", function () {
             await waitForUrl(page, "/oauth/callback");
         });
 
+        it("should successfully complete the OAuth2 Logout flow", async function () {
+            const postLogoutRedirectUri = `${TEST_CLIENT_BASE_URL}/oauth/login?logout=true`;
+
+            const { client } = await createOAuth2Client({
+                scope: "offline_access profile openid email",
+                redirectUris: [`${TEST_CLIENT_BASE_URL}/oauth/callback`],
+                postLogoutRedirectUris: [postLogoutRedirectUri],
+                accessTokenStrategy: "jwt",
+                tokenEndpointAuthMethod: "none",
+                grantTypes: ["authorization_code", "refresh_token"],
+                responseTypes: ["code", "id_token"],
+                skipConsent: true,
+            });
+
+            await setOAuth2ClientInfo(
+                page,
+                client.clientId,
+                undefined,
+                undefined,
+                undefined,
+                JSON.stringify({
+                    post_logout_redirect_uri: postLogoutRedirectUri,
+                })
+            );
+
+            await Promise.all([
+                page.waitForNavigation({ waitUntil: "networkidle0" }),
+                page.goto(`${TEST_CLIENT_BASE_URL}/oauth/login`),
+            ]);
+
+            let loginButton = await getOAuth2LoginButton(page);
+            await loginButton.click();
+
+            await waitForUrl(page, "/auth");
+
+            await toggleSignInSignUp(page);
+            const { fieldValues, postValues } = getDefaultSignUpFieldValues({ email: getTestEmail() });
+            await signUp(page, fieldValues, postValues, "emailpassword");
+
+            await waitForUrl(page, "/oauth/callback");
+
+            // Logout
+            const logoutButton = await getOAuth2LogoutButton(page, "redirect");
+            await logoutButton.click();
+
+            await waitForUrl(page, "/auth/oauth/logout");
+
+            // Click the Logout button on the provider website
+            const stLogoutButton = await waitForSTElement(page, "[data-supertokens~=button]");
+            await stLogoutButton.click();
+
+            // Ensure the final url matches the post_logout_redirect uri
+            await waitForUrl(page, "/oauth/login?logout=true", false);
+
+            await page.waitForSelector("#oauth2-token-data", { hidden: true });
+
+            // Ensure supertokens session was cleared
+            const sessionInfoRes = await fetch(`${TEST_APPLICATION_SERVER_BASE_URL}/sessioninfo`);
+            assert.strictEqual(sessionInfoRes.status, 401);
```

Comment:
AFAIK, this wouldn't fail even if the session wasn't cleared in the browser, as this fetch is running in the test runner env (node).
This needs to be checked on the frontend (e.g.: you can start the login flow again and see that it redirects to the auth screen)

Reply:
My bad!  I have updated it to check for `/auth` redirection after clicking on login. We could have also did the fetch call in `page.evaluate`, I guess.